src/hooks/usePersistentDominantColorFromImage.ts

```diff
@@ -10,52 +12,86 @@ enum State {
   failed,
 }
 
+const DEFAULT_MMKV_KEY = 'DEFAULT_MMKV_KEY';
+const NOT_DEFINED_MMKV_VALUE = 'NOT_DEFINED';
 const storage = new MMKV({
   id: STORAGE_IDS.DOMINANT_COLOR,
 });
+// settings a default value to easily check against it instead of just undefined and to make the value be always a string
+storage.set(DEFAULT_MMKV_KEY, NOT_DEFINED_MMKV_VALUE);
+
+const DEFAULT_COLOR_TO_MEASURE_AGAINST = '#333333';
+const DEFAULT_SIZE_FOR_COLOR_PROBING = 80;
 
 type Result = {
-  state: State;
   result: string | undefined;
+  state: State;
 };
 
+type ConfigOptions = {
+  colorToMeasureAgainst?: string;
+  signUrl?: boolean;
+};
 export default function usePersistentDominantColorFromImage(
```

Comment:
Could this be simplified to something similar to this?

```javascript
const [color, setColor] = React.useState(storage.get('url))

useEffect(() => {
  if (!color) {
    getDominantColorFromImage(url).then(color => setColor(color))
  }
}, [color])

return {
  color,
  loading: !color,
}
```

Reply:
Unfortunately not as simple, but I can definitely replace the useMMKVString and replace it with just state. We don't need to listen for MMKV cache changes.
The problem with your approach would be when you switch wallets. If wallet and avatar url changes we will still have the previous color.

Reply:
>If wallet and avatar url changes

Yeah, sorry, I meant my example as only a rough sketch. Like in this case you could add `walletAddress` and `avatarURL` to the `useEffect` dependency array, maybe.